tests/functional/models/test_application.py

```diff
@@ -0,0 +1,145 @@
+import unittest
+import json
+from datetime import datetime
+
+from tests.helper import TestHelper
+
+
+class TestApplication(unittest.TestCase):
+
+    helper = None
+    resin = None
+
+    @classmethod
+    def setUpClass(cls):
+        cls.helper = TestHelper()
+        cls.resin = cls.helper.resin
+
+    def tearDown(self):
+        # Wipe all apps after every test case.
+        self.helper.wipe_application()
+
+    def test_create(self):
+        # should be rejected if the device type is invalid
+        with self.assertRaises(self.helper.resin_exceptions.InvalidDeviceType):
+            self.resin.models.application.create('FooBar', 'Foo')
+
+        # should be rejected if the name has less than three characters
+        with self.assertRaises(Exception) as cm:
```

Comment:
Is this just `Exception` and not some more specific type?

Reply:
Yeah, we want to validate the error message here so a general exception is fine with me.